packages/api/v1/middleware/authenticated.ts

```diff
@@ -0,0 +1,37 @@
+import type { NextApiRequest } from 'next';
+
+import { getUserByApiToken } from '@documenso/lib/server-only/public-api/get-user-by-token';
+import type { User } from '@documenso/prisma/client';
+
+export const authenticatedMiddleware = <
+  T extends {
+    req: NextApiRequest;
+  },
+  R extends {
+    status: number;
+    body: unknown;
+  },
+>(
+  handler: (args: T, user: User) => Promise<R>,
+) => {
+  return async (args: T) => {
+    try {
+      const { authorization: token } = args.req.headers;
+
+      if (!token) {
+        throw new Error('Token was not provided for authenticated middleware');
+      }
+
+      const user = await getUserByApiToken({ token });
+
+      return await handler(args, user);
+    } catch (_err) {
+      return {
```

Comment:
This catch is only thrown when `authorization` token is invalid or null, a request without the `authorization` has some other error.

```json
HTTP/1.1 400 Bad Request

{
  "issues": [
    {
      "code": "invalid_type",
      "expected": "string",
      "received": "undefined",
      "path": [
        "authorization"
      ],
      "message": "Required"
    }
  ],
  "name": "ZodError"
}
```

Reply:
I think this is due to the request shapes requiring the authorisation header within ts-rest